### PR TITLE
Add default name for a pipeline

### PR DESCRIPTION
**Pull request by plaguss**
## Description

Generate a random name for the pipeline if one is not given by default, to simplify cognitive load:

```python
from distilabel.pipeline import Pipeline

with Pipeline() as pipeline:
    print(pipeline.name)
```

A name like the `pipeline_33prTGVu` will appear by default from now on, unless the name is given.

### CODE REVIEW

File: src/distilabel/pipeline/base.py
Comment (davidberenstein1957) [inline src/distilabel/pipeline/base.py:134]: why not use something like UUID?

Comment (plaguss) [reply src/distilabel/pipeline/base.py]: no real reason, I've just updated it as it's less code to maintain, thanks 😄

Comment (davidberenstein1957) [reply src/distilabel/pipeline/base.py]: Exactly my thoughts :)

**Comment by plaguss** [comment]
Hi @davidberenstein1957, the LLMs don't need it, the steps already have it implemented

**Review by davidberenstein1957 — APPROVED** [review]

**Comment by davidberenstein1957** [comment]
@plaguss perhaps we should update the docs where needed?

**Comment by plaguss** [comment]
Yes totally, there are some updates in  https://github.com/argilla-io/distilabel/pull/812 to the quickstart for example that should contain this (it should be here, but they are a bit mixed). Maybe we should review the docs after these 2 PRs are merged and see what to improve? Thanks for the review 🤗

**Pull request merged**
